src/authorize.ts

```diff
@@ -75,10 +75,14 @@ export const getHostedBaseUrl = (options: IEveryAuthOptions, req: express.Reques
   }
 
   // Discovery time!
-  //
+
+  // Normalize original URL by removing query
+  const i = req.originalUrl.indexOf('?');
+  const originalUrl = i < 0 ? req.originalUrl : req.originalUrl.substring(0, i);
```

Comment:
Add a test case covering this situation.

Reply:
Fixed